packages/gatsby-plugin-sharp/src/utils.js

```diff
@@ -88,33 +88,48 @@ export function rgbToHex(red, green, blue) {
     .slice(1)}`
 }
 
+const DEFAULT_FLUID_SIZE = 800
 export const calculateImageSizes = ({
   file,
+  imgDimensions,
   width,
   maxWidth,
   height,
   maxHeight,
   layout,
-  outputPixelDensities,
+  outputPixelDensities = [0.25, 0.5, 1, 2, 3],
   srcSetBreakpoints,
 }) => {
+  if (maxWidth === 0 || width === 0 || maxHeight === 0 || height === 0) {
+    throw new Error(`provided image dimensions must be positive numbers (> 0)`)
+  }
+
+  let sizes
+  const aspectRatio = imgDimensions.width / imgDimensions.height
   // Sort, dedupe and ensure there's a 1
   const densities = Array.from(new Set([1, ...outputPixelDensities])).sort()
-  let sizes
+
   if (layout === `fixed`) {
     // FIXED
     // if no width is passed, we need to resize the image based on the passed height
     const fixedDimension = width === undefined ? `height` : `width`
+    const fixedValue = fixedDimension === `height` ? height : width
+    if (fixedValue < 1) {
+      throw new Error(
+        `${fixedDimension} has to be a positive int larger than zero (> 0), now it's ${fixedValue}`
+      )
+    }
 
     if (!width) {
-      width = height * file.aspectRatio
+      width = height * aspectRatio
     }
 
     sizes = densities
+      .filter(size => size >= 1) // remove smaller densities because fixed images don't need them
       .map(density => density * width)
-      .filter(size => size <= width)
+      .filter(size => size <= imgDimensions.width)
```

Comment:
I know this is less than or equal but it looks like a reverse arrow and almost broke my brain haha